FLOE-198: Add UIO to the Feedback Tool Demo

jobara opened the pull request:
Added UIO to the feedback tool demo. 

http://issues.fluidproject.org/browse/FLOE-198

jobara committed:
FLOE-198: Updated to Infusion 1.5

Also added in UI Options, which will be needed for the Feedback tool demo.

jobara committed:
FLOE-198: Updated VERSION.md

Added removal of README.md from infusion

jobara committed:
FLOE-198: Adding in FSS as it is needed by UIO

jobara committed:
FLOE-198: Adding UIO to feedback demo

jobara committed:
FLOE-198: Fixing the overview panel

Fixing up the overview panel after the Infusion upgrade

jobara committed:
FLOE-198: Updated comment

jobara committed:
FLOE-198: Removed commented out config

jobara committed:
FLOE-198: Combining document readys

jobara committed:
Merge branch 'FLOE-193' into FLOE-198

Conflicts:
	demos/feedback/css/feedbackDemo.css
	demos/html/feedback.html

jobara committed:
FLOE-198: Re-added UIO to feedback demo

Re-added UIO to feedback demo after merge, since the merge removed the changes. Updated the styles to better align the UIO tab with the content.

jobara committed:
Merge branch 'FLOE-193' into FLOE-198

jobara committed:
FLOE-198: Added contrast themes for icons.

jobara committed:
Merge branch 'master' into FLOE-198

jobara committed:
FLOE-198: Fixed line spacing in the feedback demo

Set a default line-height on the body and set <p> in the demo markup to inherit. This overrides foundation's default line-height and allows the line spacing enactor to set the line-height.

cindyli commented:
Some file permissions are not 644.

cindyli commented:
Click "show display preferences" to open up UIO, the "reset" button overlaps the header text of the demo. The overlapping is worse when the preference "enlarge buttons" is turned on or on a small screen.

cindyli commented on demos/feedback/html/index.html:
```diff
@@ -56,6 +92,25 @@ <h2 class="gpii-feedbackDemo-page-title">Climate Change Impacts</h2>
+                        options: {
+                            iframeRenderer: {
+                                markupProps: {
+                                    "class": "flc-iframe",
```
Can this line be removed since the value is the default one that's already defined in prefs framework?

cindyli commented:
When high (or low) contrast theme is applied, the circles around preference buttons on the feedback top toolbar disappears, but the circles for happy/sad face and request buttons have color changed properly.

cindyli commented:
When text size increases, the size of the video container on the feedback demo page doesn't increase accordingly.

cindyli commented:
1. Turn on "table of contents" in UIO;
2. Reload the page;
3. Open up firebug console; 
4. Click "show display preferences" to open up UIO.

A javascript error occurs:
"NetworkError: 404 Not Found - http://localhost/metadata/demos/components/tableOfContents/html/TableOfContents.html?_=1405452893640"

Seems the UIO iframe panel is looking for table of content template at a wrong location. This template is inside the infusion lib - src/lib/infusion/components/tableOfContents/html/TableOfContents.html

jobara committed:
FLOE-198: Fixing file permissions

jobara committed:
FLOE-198: removed class option for instantiation

jobara commented:
I talked to Dana and Joanna about the Feedback Tool design and UIO. They had some suggestions which relate to the comments from @cindyli above. 
- The display preferences button should be smaller (see: [outside-in](http://outside-in.idrc.ocadu.ca/) site)
- Joanna will provide mockups for the preference icons when contrast themes are applied
- The video element should scale with the text
- The Feedback Tool should push down the page content if the toolbar wraps and expands.

jobara committed:
FLOE-198: Video container scales with text

jobara committed:
FLOE-198: Converted em to rem for demo css

jobara committed:
FLOE-198: Converted Feedback css to rem

Updated the style of the demo to fit the new size of the feedback tool.

jobara committed:
FLOE-198: Smaller display preferences button

jobara committed:
FLOE-198: Fixed icon alignment

The icon alignment had broken in IE with the use of rems. The fix was actually to greatly simplify the styling used to setup the icon-fonts.

jobara committed:
FLOE-198: Further tweaks on display preferences

Made the display preferences tab smaller, and addressed some more contrast styling issues.

jobara committed:
FLOE-198: Cleaning markup and classes

jobara committed:
FLOE-198: Updated the styles of the badges

Added styles for the various states of the preference icons, both with and without contrast themes applied.

jobara commented:
The issue that @cindyli described with the ToC template above was an issue in Infusion. http://issues.fluidproject.org/browse/FLUID-5474

jobara commented:
I haven't fixed the issue of the feedback tool wrapping at small screens. I think we may want to provide a responsive design for this instead.

jobara committed:
FLOE-198: Adjusted the floating of icons

The icons now float as a set for the preference icons and the control icons.

cindyli commented:
@jobara, thanks for fixing those. Now the icons look great even with high contrast themes.

A minor issue I noticed is if you use "Verdana" font, maximize line spacing, enlarge buttons, the reset button would overlap the header text.

jobara committed:
FLOE-198: Set a line-height on the UIO Button

Set a line-height on UIO's display preferences button to prevent it from increasing in size as line-height increased. This prevents it from overlapping the header.

jobara commented:
@cindyli ready for another round of review. I've addressed the overlap issue when line spacing is set. I've also filed a jira for the feedback tool overlapping the header and the UIO button when the icons don't fit on a single row. http://issues.fluidproject.org/browse/FLOE-200

cindyli commented:
Merged into the master branch @ 017b8c8163a3433bc3ad6282c4ed9088cbede38b
